packages/neuron-wallet/package.json

```diff
@@ -55,6 +55,8 @@
     "sha3": "2.0.7",
     "sqlite3": "4.1.1",
     "subleveldown": "^4.1.4",
+    "worker-threads.js": "1.6.5",
```

Comment:
Is it necessary to introduce a third-party package?

Reply:
By using this library, the code for master/child process communication is much clearer than the built-in pub/sub pattern, and the same codebase can be changed by changing a line of code to start `worker thread` or `child_process`.

Reply:
> By using this library, the code for master/child process communication is much clearer than the built-in pub/sub pattern, and the same codebase can be changed by changing a line of code to start `worker thread` or `child_process`.

Can we implement it in this project? We are going to reduce deps since they are out of control.

Reply:
Sure, we can. I'll remove this dependency and implement it myself.
BTW, How to define a library that is "out of control", is there a guideline?

Reply:
> Sure, we can. I'll remove this dependency and implement it myself.
> BTW, How to define a library that is "out of control", is there a guideline?

There're no explicit rules to decide if a dep is acceptable, but the strategy made before is conservative(and some relative rules can be found in wiki/package-dependency-upgrade-strategy in the internal repo).

- The best choice is not to use a third-party package.
- The second one is to use a package having reputations.
- The third one is to use a package that is generally used and implement it ourselves once we can.

The 2nd. and 3rd. are not ideal though since famous packages or widely used packages are also fragile, such as `event-stream` which was actually attacked and had an incredible effect.

Reply:
Thank you for your thorough explanation. 👍